widgets/common/src/deploymentsView/header/RunWorkflowModal.tsx

```diff
@@ -46,59 +49,57 @@ const RunWorkflowModal: FunctionComponent<RunWorkflowModalProps> = ({ filterId,
     const { errors, setErrors, clearErrors, setMessageAsError } = useErrors();
-    const [workflowId, setWorkflowId, resetWorkflowId] = useResettableState<string>('');
+    const [workflowId, setWorkflowId, resetWorkflowId] = useResettableState('');
     const [workflows, setWorkflows, resetWorkflows] = useResettableState<Workflow[]>([]);
-    const [loadingMessage, setLoadingMessage, resetLoadingMessage] = useResettableState('');
+    const [loadingMessage, setLoadingMessage, turnOffLoading] = useResettableState('');
     const workflowsOptions = useMemo(() => getWorkflowsOptions(workflows), [workflows]);
 
+    const searchActions = new Stage.Common.SearchActions(toolbox);
+
     useEffect(() => {
         clearErrors();
         resetWorkflowId();
         resetWorkflows();
         unsetExecutionGroupStarted();
         setLoadingMessage(modalT('messages.fetchingWorkflows'));
 
-        toolbox
-            .getManager()
-            .doGet('/workflows', {
-                _filter_id: filterId
-            })
+        searchActions
+            .doListWorkflows(filterRules)
             .then((data: WorkflowsResponse) => setWorkflows(data.items))
             .catch(setMessageAsError)
-            .finally(resetLoadingMessage);
-    }, [filterId]);
+            .finally(turnOffLoading);
+    }, []);
 
-    function runWorkflow() {
+    async function runWorkflow() {
         if (!workflowId) {
             setErrors({ error: modalT('errors.noWorkflowError') });
             return;
         }
 
-        setLoadingMessage(modalT('messages.creatingDeploymentGroup'));
-        const deploymentGroupsActions = new Stage.Common.DeploymentGroupsActions(toolbox);
-        const groupId = `BATCH_ACTION_${new Date().toISOString()}`;
+        try {
+            setLoadingMessage(modalT('messages.creatingDeploymentGroup'));
+            const groupId = getGroupIdForBatchAction();
+            const deploymentGroupsActions = new Stage.Common.DeploymentGroupsActions(toolbox);
+            await deploymentGroupsActions.doCreate(groupId, { filter_rules: filterRules });
 
-        deploymentGroupsActions.doCreate(groupId, filterId!).then(_deploymentGroup => {
             setLoadingMessage(modalT('messages.startingExecutionGroup'));
             const executionGroupsActions = new Stage.Common.ExecutionGroupsActions(toolbox);
+            await executionGroupsActions.doStart(workflowId, groupId);
 
-            return executionGroupsActions.doStart(workflowId!, groupId).then(_executionGroup => {
-                toolbox.getEventBus().trigger('deployments:refresh');
-                toolbox.getEventBus().trigger('executions:refresh');
-                resetLoadingMessage();
-                setExecutionGroupStarted();
-            });
-        });
-    }
+            toolbox.getEventBus().trigger('deployments:refresh');
+            toolbox.getEventBus().trigger('executions:refresh');
```

Comment:
After my recent change it can be shortened to:
```suggestion
            toolbox.getEventBus().trigger('deployments:refresh').trigger('executions:refresh');
```

Reply:
Sure, done.